dependency.lic

```diff
@@ -498,12 +493,69 @@ class ScriptManager
     http.verify_mode = OpenSSL::SSL::VERIFY_NONE
 
     request = Net::HTTP::Get.new(uri.request_uri)
-    request['Authorization'] = "token #{Base64.decode64(@combine_character_deletion)}"
+    request['Authorization'] = @request_authorization
 
     response = http.request(request)
     JSON.parse(response.body)
   end
 
+  def manage_github_token
+    # This method manages some checking of the GitHub token we'll be using. We allow a personal
+    # GitHub token to be specified in the /data folder, or we can use the general Lich token.
+    # If using a personal GitHub token, we'll verify the token is valid, and we'll also do some
+    # checking of the rate limit for the token we're using. If the rate limit is above a certain
+    # threshold, we'll warn but won't fail, as Lich can still function when the rate limit is
+    # exceeded, albeit with some reduced functionality.
+    #  Returns: request['Authorization'] token string
+
+    github_token = [
+      'Z2hwX21XN1',
+      'V0RkdORXdGYlpmVDF1S3AzM2I1TV',
+      'J5bHlTNjBXc09VNw==',
+      '\n'
+    ].join('')
+
+    # Set up our rate limit check URL and items
+    url = URI("https://api.github.com/rate_limit")
+    http = Net::HTTP.new(url.host, url.port)
+    http.use_ssl = true
+    request = Net::HTTP::Get.new(url)
+
+    # Check if the user has a githubtoken.txt file and use the token in that file if so. If they
+    # do have a personal token specified, verify valid. Else, use general token.
+    if File.exist?("#{DATA_DIR}/githubtoken.txt")
+      echo "Using personal GitHub token." if @debug
+      request['Authorization'] = "token #{File.read("#{DATA_DIR}/githubtoken.txt")}"
```

Comment:
```suggestion
      request['Authorization'] = "token #{File.read("#{DATA_DIR}/githubtoken.txt").strip}"
```

Reply:
First time suggesting code... sorry. I don't know why it keeps breaking the line.

Reply:
Added, still works in Windows.